### ISSUE TITLE
Training "pause" and CPU runs at 100 when using visual observations

### ISSUE THREAD

**Issue by KristianLN** [Unity-Technologies/ml-agents#2430]
Hi,

I'm experimenting with incorporating visual observation in my agent but my CPU doesn't seem to be happy with the use of visual observations.

I experience a sort of dropout, where the application freezes for a x amount of time, and the CPU utilization goes up to 100, before training goes on and continue. This repeats continuously throughout training. 

Is it because my CPU isn't strong enough, or is there something I can do about it?

CPU: Intel(R) Core(TM) i5-7300HQ CPU @ 2.50 GHz

I'm currently experimenting with different hyper parameters to handle the dropouts:

**Common specifications:**
One continuous action, 126 vector observations, one camera for visual observation (front of the agent, colour)
Default hyper parameters unless stated otherwise below.
Stacked observations: 1
TimeScale: 50 (I tried with 100 but there was no advantages, in terms of training time, of that compared to 50)

Test runs:
0: All default hyper parameters, and else stated as above: dropout time of around 120 seconds.
1: Buffer lowered to 5120 from initial 10240; dropout time lowered to around 70 seconds.
2: Buffer lowered further to 2560, batch lowered to 512; dropout time lowered to around 40 seconds.
3: Buffer 2560, batch lowered to 256; Dropout still around 40 seconds.
4: Buffer 2560, batch increased to 512 again, NumLayers lowered to 1 from 2 and numEpoch lowered to 2 from 3; dropout time around 30 seconds.

All of this is even without considering the effect on the performance of the agent ...

I would appreciate any ideas to further decrease the dropout time or ideally completely remove it.

**Comment by ervteng** [comment]
Hi @KristianLN, this "dropout" time is the time when the model is being trained. When the simulation is running, the trainer is gathering experiences (images) and collecting them in a buffer, then the TensorFlow code performs a model update. So it isn't lost time! To decrease the dropout time I'd suggest using a computer with CUDA + GPU, and installing tensorflow-gpu (especially for visual observations).

**Comment by KristianLN** [comment]
Hi @ervteng 
Thank you!! That makes sense why I see improvements when I play around with the  buffer/batch den.
I'm almost certain I didn't experience the dropout time when I only used vector observations, but of cause images is a completely different size to process.

Thank you once again!
